qiskit_aer/primitives/estimator_v2.py

```diff
@@ -0,0 +1,146 @@
+      Default: {}.
+    """
+
+    def __init__(
+        self,
+        *,
+        options: dict | None = None,
+    ):
+        """
+        Args:
+            options: The options to control the default precision (``default_precision``),
+                the backend options (``backend_options``), and
+                the runtime options (``run_options``).
+        """
+        self._options = Options(**options) if options else Options()
+        method = "density_matrix" if "noise_model" in self.options.backend_options else "automatic"
+        self._backend = AerSimulator(method=method, **self.options.backend_options)
+
+    @property
+    def options(self) -> Options:
+        """Return the options"""
+        return self._options
+
+    def run(
+        self, pubs: Iterable[EstimatorPubLike], *, precision: float | None = None
+    ) -> PrimitiveJob[PrimitiveResult[PubResult]]:
+        if precision is None:
+            precision = self._options.default_precision
+        coerced_pubs = [EstimatorPub.coerce(pub, precision) for pub in pubs]
+        self._validate_pubs(coerced_pubs)
+        job = PrimitiveJob(self._run, coerced_pubs)
+        job._submit()
+        return job
+
+    def _validate_pubs(self, pubs: list[EstimatorPub]):
+        for i, pub in enumerate(pubs):
+            if pub.precision < 0.0:
+                raise ValueError(
+                    f"The {i}-th pub has precision less than 0 ({pub.precision}). ",
+                    "But precision should be equal to or larger than 0.",
+                )
+
+    def _run(self, pubs: list[EstimatorPub]) -> PrimitiveResult[PubResult]:
+        return PrimitiveResult([self._run_pub(pub) for pub in pubs])
+
+    def _run_pub(self, pub: EstimatorPub) -> PubResult:
+        circuit = pub.circuit.copy()
+        observables = pub.observables
+        parameter_values = pub.parameter_values
+        precision = pub.precision
+
+        # calculate broadcasting of parameters and observables
+        param_shape = parameter_values.shape
+        param_indices = np.fromiter(np.ndindex(param_shape), dtype=object).reshape(param_shape)
+        bc_param_ind, bc_obs = np.broadcast_arrays(param_indices, observables)
+
+        parameter_binds = {}
+        param_array = parameter_values.as_array(circuit.parameters)
+        parameter_binds = {p: param_array[..., i].ravel() for i, p in enumerate(circuit.parameters)}
+
+        # save expval
+        paulis = {pauli for obs_dict in observables.ravel() for pauli in obs_dict.keys()}
+        for pauli in paulis:
+            circuit.save_expectation_value(
+                Pauli(pauli), qubits=range(circuit.num_qubits), label=pauli
+            )
+        result = self._backend.run(
+            circuit, parameter_binds=[parameter_binds], **self.options.run_options
+        ).result()
+
+        # calculate expectation values (evs) and standard errors (stds)
+        rng = np.random.default_rng(self.options.run_options.get("seed_simulator"))
```

Comment:
I'm wondering that `rng` should be shared in a ` _run()` call.

Reply:
I think implementation of rng is as same as that of qiskit.primitives https://github.com/Qiskit/qiskit/blob/090b2b193d6de88017e26d99c0ac379b85382395/qiskit/primitives/statevector_estimator.py#L142

Reply:
Yes. This is the point where we decided to make this specification for maintainability when we implemented SamplerV2.